File: backend/src/prechecks/concrete/cluster/cluster-health.precheck.ts
import { ElasticClient } from "../../../clients/elastic.client";
import { BaseClusterPrecheck } from "../../base/base-cluster-precheck";
import { ExecutionMode, PrecheckType } from "../../types/enums";
import { PrecheckExecutionRequest, ClusterContext } from "../../types/interfaces";

export class ClusterHealthPrecheck extends BaseClusterPrecheck {
	constructor() {
		super({
			id: "elasticsearch_cluster_health_check",
			name: "Cluster health check",
			type: PrecheckType.CLUSTER,
			mode: ExecutionMode.CODE,
		});
	}
	protected async runForContext(request: PrecheckExecutionRequest<ClusterContext>): Promise<void> {
		const client = await ElasticClient.buildClient(request.cluster.clusterId);
		const elasticsearchClient = client.getClient();
		const health = await elasticsearchClient.cluster.health();
		const isHealthy = health.status === "green";

		const message = `Cluster health status: '${health.status}'. Expected: 'green'.`;

		await this.addLog(request, message);

		if (!isHealthy) {
			throw new Error(`Cluster health check failed. ${message}`);
		}
	}
}

File: server/modules/precheck/src/main/java/co/hyperflex/precheck/concrete/cluster/ClusterHealthPrecheck.java
package co.hyperflex.precheck.concrete.cluster;

import co.hyperflex.precheck.contexts.ClusterContext;
import co.hyperflex.precheck.core.BaseClusterPrecheck;
import org.slf4j.Logger;
import org.springframework.stereotype.Component;

@Component
public class ClusterHealthPrecheck extends BaseClusterPrecheck {

  @Override
  public String getName() {
    return "Cluster health check";
  }

  @Override
  public void run(ClusterContext context) {
    Logger logger = context.getLogger();
    try {
      String status = context.getElasticClient().getHealthStatus();
      boolean isHealthy = "green".equalsIgnoreCase(status);

      logger.info("Cluster health status: '{}'. Expected: 'green'.", status);

      if (!isHealthy) {
        logger.error("Cluster health check failed");
        throw new RuntimeException();
      }
    } catch (Exception e) {
      logger.error("Failed to check cluster health", e);
      throw new RuntimeException();
    }
  }
}
